ansible/roles/test/files/ptftests/vlan_test.py

```diff
@@ -0,0 +1,268 @@
+import ast
+import json
+import logging
+import subprocess
+
+from collections import defaultdict
+from ipaddress import ip_address, ip_network
+
+import ptf
+import ptf.packet as scapy
+import ptf.dataplane as dataplane
+
+from ptf import config
+from ptf.base_tests import BaseTest
+from ptf.testutils import *
+from ptf.mask import Mask
+
+class VlanTest(BaseTest):
+    def __init__(self):
+        BaseTest.__init__(self)
+        self.test_params = test_params_get()
+    #--------------------------------------------------------------------------
+
+    def log(self, message):
+        logging.info(message)
+    #--------------------------------------------------------------------------
+
+    def shell(self, cmds):
+        sp = subprocess.Popen(cmds, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = sp.communicate()
+        rc = sp.returncode
+
+        return stdout, stderr, rc
+    #--------------------------------------------------------------------------
+
+    def setUp(self):
+        self.vlan_ports_list = ast.literal_eval(self.test_params["vlan_ports_list"])
+        self.vlan_intf_list = ast.literal_eval(self.test_params["vlan_intf_list"])
+        self.router_mac = self.test_params["router_mac"]
+
+        for vlan_port in self.vlan_ports_list:
+            vlan_port["pvid"] = int(vlan_port["pvid"])
+            vlan_port["port_index"] = int(vlan_port["port_index"])
+
+        self.dataplane = ptf.dataplane_instance
+        self.test_params = test_params_get()
+        self.setUpArpResponder()
+        self.log("Start arp_responder")
+        self.shell(["supervisorctl", "start", "arp_responder"])
+
+        self.log("Create VLAN intf")
+        for vlan_port in self.vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) != vlan_port["pvid"]:
+                    self.shell(["ip", "link", "add", "link", "eth%d"%vlan_port["port_index"], 
+                                "name", "eth%d.%s"%(vlan_port["port_index"], permit_vlanid), 
+                                "type", "vlan", "id", str(permit_vlanid)])
+                    self.shell(["ip", "link", "set", 
+                                "eth%d.%s"%(vlan_port["port_index"], permit_vlanid), "up"])
+
+        logging.info("VLAN test starting ...")
+        pass
+    #--------------------------------------------------------------------------
+
+    def setUpArpResponder(self):
```

Comment:
in order for the test to make, every port in the ptf conainer needs to have a different mac. That is not the setup by default, you need to run this script. Can you add in the test?

https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/files/helpers/change_mac.sh